plugins/in_tail/tail_fs_stat.c

```diff
@@ -119,15 +119,21 @@ static int tail_fs_check(struct flb_input_instance *ins,
             continue;
         }
 
+        int64_t size_delta = st.st_size - file->size;
+        if (size_delta != 0) {
+            file->size = st.st_size;
```

Comment:
This is the only place in this PR where the change is not restricted to the method scope and might have any impact outside the method. This file->size assignment was not  here before (and it could be removed as we only need size_delta to detect truncation). I introduced this only for consistency with the other implementations.